### ISSUE TITLE
✨ Add "Remove" and "Re-instate" dialogs to Pool Candidate page

### ISSUE THREAD

**Issue by tristan-orourke** [GCTC-NTGC/gc-digital-talent#9198]
## ✨ Feature

From the Pool Candidate page, admins should be able to "remove" candidates, or later reinstate them.

Note that "removal" is different from "disqualification", though it also stops candidates from appearing in searches. It doesn't imply that the user has failed any assessments, and can be done to users even after Qualification (or disqualification for that matter) but not to users who are already Placed.

## 🕵️ Details

- [ ] The "Remove candidate" button opens the Remove candidate dialog, where candidate can be set to Removed, with accompanying reason
- [ ] If the candidate is currently removed, "Remove candidate" and "Record final decision" buttons are replaced with a "Reinstate candidate button" which becomes the main call to action, and opens the Revert Removed decision dialog
- [ ] Use the mutations defined in #9191

The PoolCandidateStatuses which count as "removed" are
- SCREENED_OUT_NOT_INTERESTED
- SCREENED_OUT_NOT_RESPONSIVE
- QUALIFIED_UNAVAILABLE
- QUALIFIED_WITHDREW
- REMOVED

## 🎨 Design File

https://www.figma.com/file/em5ArPr044YaIe1Vb1zCZz/Admin---Processes%2FPools%2FRecruitments---Record-of-Decision?type=design&node-id=8165-21175&mode=design&t=o2pTls0JmG0q1lfN-0

### 🧑‍🎨 Designer

@Jerryescandon 

### Actions buttons, which need to change contextually
![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/4313717/163f0d44-a6c6-4994-8861-dd9aa84ec7d2)

### Remove candidate
![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/4313717/7b8b31ba-43a3-4946-849e-e86d13314a67)

### Reinstate candidate
![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/4313717/a9b07cb5-b289-4bab-a2c4-0884d43720c0)

## 🙋‍♀️ Proposed Implementation

(optional) Do you have a proposed implementation?

## 🌎 Localization

(optional) Provide any new copy along with translations available.

## ✅ Acceptance Criteria

- [ ] The "Remove candidate" button opens the Remove candidate dialog, where candidate can be set to Removed, with accompanying reason
- [ ] If the candidate is currently removed, "Remove candidate" and "Record final decision" buttons are replaced with a "Reinstate candidate button" which becomes the main call to action, and opens the Revert Removed decision dialog
- [ ] Use the mutations defined in #9191

**Comment by tristan-orourke** [comment]
@Jerryescandon We're removing the "Language testing was unsuccessful option"

**Comment by esizer** [comment]
Just a note, we can use the `REMOVED_STATUSES` constant if you filter out expired. Unless we do want to allow reinstating a candidate who has expired 🤔 

https://github.com/GCTC-NTGC/gc-digital-talent/blob/24ec09de22d793eef4f85bfef44c2a85c873b7f1/apps/web/src/constants/poolCandidate.ts#L18-L25

**Comment by tristan-orourke** [comment]
Okay, this got me thinking, and I think we _do_ want to be able to reinstate a candidate who has the expired _STATUS_.

It's important to remember that the Expired status is a legacy status, from before we used expiry dates properly. If an admin user manually gave someone an expired status, there should be a way to change that (ie this re-instate interaction). For the normal process of expiry, where they remain in a QUALIFIED_AVAILABLE status but the expiry date passes, you'd change the expiry date to un-expire them.

**Comment by esizer** [comment]
> Okay, this got me thinking, and I think we _do_ want to be able to reinstate a candidate who has the expired _STATUS_.

Yeah, that all makes sense to me! I noticed it was super close so I was hoping we could just use the existing values :laughing:
